### PR TITLE
Exported new functions for txPreprocess - prepared for sovereign chain

### PR DESCRIPTION
**Pull request by sasurobert**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- 
- 
- 
  
## Proposed Changes
- 
- 
- 

## Testing procedure
- 
- 
-

### CODE REVIEW

File: process/block/preprocess/export.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+// ProcessTxsFromMe exported function
+func (txs *transactions) ProcessTxsFromMe(
+	body *block.Body,
+	haveTime func() bool,
+	randomness []byte,
+) (block.MiniBlockSlice, map[string]struct{}, error) {
+	return txs.processTxsFromMe(body, haveTime, randomness)
+}
+
+// CreateScheduledMiniBlocks only the scheduled miniblocks
```
Comment (SebastianMarian) [inline process/block/preprocess/export.go:27]: Reformulate the comment

File: process/block/preprocess/export.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		)
+		return make(block.MiniBlockSlice, 0), nil
+	}
+
+	log.Debug("elapsed time to computeSortedTxs",
+		"num txs", len(sortedTxs),
+		"time [s]", elapsedTime,
+	)
+
+	if txs.blockTracker.ShouldSkipMiniBlocksCreationFromSelf() {
```
Comment (SebastianMarian) [inline process/block/preprocess/export.go:58]: L58-L61 could be removed, not applicable for side-chain as this method uses meta chain to check the global stuck

File: process/block/preprocess/transactions.go [hunk 3]
```diff
@@ -61,6 +61,8 @@ type transactions struct {
 	emptyAddress                 []byte
 	txTypeHandler                process.TxTypeHandler
 	scheduledTxsExecutionHandler process.ScheduledTxsExecutionHandler
+
+	scheduledTXContinueFunc func(isShardStuck func(uint32) bool, wrappedTx *txcache.WrappedTransaction, mapSCTxs map[string]struct{}, mbInfo *createScheduledMiniBlocksInfo) (*transaction.Transaction, *block.MiniBlock, bool)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/block/preprocess/transactions.go:65]: left as it is. it is shorter.

File: process/block/preprocess/transactions.go [hunk 4]
```diff
@@ -630,11 +629,34 @@ func (txs *transactions) processTxsFromMe(
...
+		return false
+	}
+	isMaxBlockSizeReachedFalse := func(int, int) bool {
+		return false
+	}
+	haveAdditionalTimeFalse := func() bool {
+		return false
 	}
 
 	scheduledMiniBlocks, err := txs.createAndProcessScheduledMiniBlocksFromMeAsValidator(
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/block/preprocess/transactions.go]: done

File: process/block/preprocess/export.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+		log.Debug("CreateAndProcessMiniBlocks global stuck")
+		return make(block.MiniBlockSlice, 0), nil
+	}
+
+	sortedTxsForScheduled := append(sortedTxs, remainingTxsForScheduled...)
+	sortedTxsForScheduled, _ = txs.prefilterTransactions(nil, sortedTxsForScheduled, 0, gasBandwidth)
+	txs.sortTransactionsBySenderAndNonce(sortedTxsForScheduled, randomness)
+
+	haveAdditionalTime := process.HaveAdditionalTime()
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/block/preprocess/export.go]: done

File: process/block/preprocess/export.go [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+		sortedTxsForScheduled,
+		make(map[string]struct{}),
+	)
+	if err != nil {
+		log.Debug("createAndProcessScheduledMiniBlocksFromMeAsProposer", "error", err.Error())
+		return make(block.MiniBlockSlice, 0), nil
+	}
+
+	return scheduledMiniBlocks, nil
+}
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/block/preprocess/export.go]: done

File: process/block/preprocess/export.go [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+)
+
+// SetScheduledTXContinueFunc sets a new scheduled tx verifier function
+func (txs *transactions) SetScheduledTXContinueFunc(newFunc func(isShardStuck func(uint32) bool, wrappedTx *txcache.WrappedTransaction, mapSCTxs map[string]struct{}, mbInfo *createScheduledMiniBlocksInfo) (*transaction.Transaction, *block.MiniBlock, bool)) {
+	if newFunc != nil {
+		txs.scheduledTXContinueFunc = newFunc
+	}
+}
+
+// ProcessTxsFromMe exported function
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/block/preprocess/export.go]: done

File: process/block/preprocess/export.go (hunk 1 above)
Comment (sasurobert) [reply process/block/preprocess/export.go]: done

File: process/block/preprocess/export.go [hunk 8]
```diff
@@ -0,0 +1,74 @@
+package preprocess
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/storage/txcache"
+	"time"
```
Comment (iulianpascalau) [inline process/block/preprocess/export.go:7]: go imports

File: process/block/preprocess/transactions.go [hunk 9]
```diff
@@ -630,14 +629,37 @@ func (txs *transactions) processTxsFromMe(
...
+	body *block.Body,
+	haveTime func() bool,
+	randomness []byte,
+) error {
+	calculatedMiniBlocks, mapSCTxs, err := txs.processTxsFromMe(body, haveTime, randomness)
+	if err != nil {
+		return err
+	}
+
+	isShardStuckFalse := func(uint32) bool {
```
Comment (iulianpascalau) [inline process/block/preprocess/transactions.go:652]: these 3 functions could have been declared as vars as we might reuse them in other functions

File: process/block/preprocess/export.go [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+)
+
+// SetScheduledTXContinueFunc sets a new scheduled tx verifier function
+func (txs *transactions) SetScheduledTXContinueFunc(newFunc func(isShardStuck func(uint32) bool, wrappedTx *txcache.WrappedTransaction, mapSCTxs map[string]struct{}, mbInfo *createScheduledMiniBlocksInfo) (*transaction.Transaction, *block.MiniBlock, bool)) {
```
Comment (iulianpascalau) [inline process/block/preprocess/export.go:11]: some unit tests would be great

File: process/block/preprocess/export.go [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+	log.Debug("elapsed time to computeSortedTxs",
+		"num txs", len(sortedTxs),
+		"time [s]", elapsedTime,
+	)
+
+	sortedTxsForScheduled := append(sortedTxs, remainingTxsForScheduled...)
+	sortedTxsForScheduled, _ = txs.prefilterTransactions(nil, sortedTxsForScheduled, 0, gasBandwidth)
+	txs.sortTransactionsBySenderAndNonce(sortedTxsForScheduled, randomness)
+
+	haveAdditionalTime := func() bool { return false }
```
Comment (iulianpascalau) [inline process/block/preprocess/export.go:61]: this could have been: `haveAdditionalTimeFalse`

**Review by SebastianMarian — DISMISSED** [review]

File: process/block/preprocess/export.go (hunk 8 above)
Comment (sasurobert) [reply process/block/preprocess/export.go:7]: done

File: process/block/preprocess/export.go (hunk 11 above)
Comment (sasurobert) [reply process/block/preprocess/export.go:61]: done

File: process/block/preprocess/transactions.go (hunk 9 above)
Comment (sasurobert) [reply process/block/preprocess/transactions.go:652]: done

**Review by iulianpascalau — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**